### PR TITLE
Update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by marthacryan**
Updates documentation based on issues that I ran into building today.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -77,12 +77,14 @@ We use the following [labels](https://github.com/plotly/plotly.js/labels) to tra
 #### Prerequisites
 
 - git
-- [node.js](https://nodejs.org/en/). We recommend using node.js v16.x.
+- [node.js](https://nodejs.org/en/). We recommend using node.js v20.x.
```
Comment (archmoj) [inline CONTRIBUTING.md:80]: Node v16 should be fine. In fact that's the version we use today for most of our tests as well as the release.
Even `node v12` is still supported in our build processes and tested here: https://github.com/plotly/plotly.js/blob/8c47c16daaa2020468baf9376130e085a4f01ec6/.circleci/config.yml#L13

What was the problem you encountered with node v16?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -77,12 +77,14 @@ We use the following [labels](https://github.com/plotly/plotly.js/labels) to tra
...
   Upgrading and managing node versions can be easily done using
   [`nvm`](https://github.com/creationix/nvm) or its Windows alternatives.
 - [`npm`](https://www.npmjs.com/) v7.x and up to ensure that the
   [`package-lock.json`](https://docs.npmjs.com/files/package-lock.json) file is
   used and updated correctly.
+- [`python3`](https://www.python.org/downloads/)
+Note: for M1 mac users specifically, you might need to install some [extra dependencies](https://github.com/Automattic/node-canvas/issues/1733#issuecomment-761703018).
```
Comment (archmoj) [inline CONTRIBUTING.md:87]: Wondering if you could possibly add a test for mac users?

**Comment by gvwilson** [comment]
@marthacryan if you can please address @archmoj 's comments, we can get this one merged - thank you

File: CONTRIBUTING.md (hunk 1 above)
Comment (gvwilson) [reply CONTRIBUTING.md:80]: @marthacryan rekindling - what was the issue w/ node v16?

File: CONTRIBUTING.md (hunk 2 above)
Comment (gvwilson) [reply CONTRIBUTING.md:87]: ping @marthacryan

File: CONTRIBUTING.md (hunk 1 above)
Comment (marthacryan) [reply CONTRIBUTING.md:80]: I'm not sure what the issue was with node v16 but I'm no longer having it. I'll remove this change!

**Review by archmoj — APPROVED** [review]
Thanks very much for the PR.
:dancer:

**Pull request merged**